src/main/java/mate/academy/service/impl/ShoppingCartServiceImpl.java

```
@@ -0,0 +1,52 @@
+package mate.academy.service.impl;
+
+import java.util.NoSuchElementException;
+import mate.academy.dao.ShoppingCartDao;
+import mate.academy.dao.TicketDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.MovieSession;
+import mate.academy.model.ShoppingCart;
+import mate.academy.model.Ticket;
+import mate.academy.model.User;
+import mate.academy.service.ShoppingCartService;
+
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    @Inject
+    private ShoppingCartDao shoppingCartDao;
+    @Inject
+    private TicketDao ticketDao;
+
+    @Override
+    public void addSession(MovieSession movieSession, User user) {
+        Ticket ticket = new Ticket();
+        ticket.setUser(user);
+        ticket.setMovieSession(movieSession);
+        ShoppingCart shoppingCart = shoppingCartDao.getByUser(user).orElseThrow(
```

it is better to fail fast, you can first get cart by user and than make all other logic